dpdispatcher/openapi_context.py

```diff
@@ -0,0 +1,252 @@
+from dpdispatcher.JobStatus import JobStatus
+
+# 引入一个sqlite数据库，存储job相关信息
+DP_CLOUD_SERVER_HOME_DIR = os.path.join(
+    os.path.expanduser("~"), ".dpdispatcher/", "dp_cloud_server/"
+)
+
+
+class OpenAPIContext(BaseContext):
+    def __init__(
+        self,
+        local_root,
+        remote_root=None,
+        remote_profile={},
+        *args,
+        **kwargs,
+    ):
+        self.init_local_root = local_root
+        self.init_remote_root = remote_root
+        self.temp_local_root = os.path.abspath(local_root)
+        self.remote_profile = remote_profile
+        self.client = Client()
+        self.storage = Storage(client=self.client)
+        self.job = Job(client=self.client)
+        self.util = Util()
+        self.jgid = None
+
+    @classmethod
+    def load_from_dict(cls, context_dict):
+        local_root = context_dict.get("local_root", "./")
+        remote_root = context_dict.get("remote_root", None)
+        remote_profile = context_dict.get("remote_profile", {})
+
+        bohrium_context = cls(
+            local_root=local_root,
+            remote_root=remote_root,
+            remote_profile=remote_profile,
+        )
+        return bohrium_context
+
+    def bind_submission(self, submission):
+        self.submission = submission
+        self.local_root = os.path.join(self.temp_local_root, submission.work_base)
+        self.remote_root = "."
+
+        self.submission_hash = submission.submission_hash
+
+        self.machine = submission.machine
+
+    def _gen_object_key(self, job, zip_filename):
+        if hasattr(job, "upload_path") and job.upload_path:
+            return job.upload_path
+        else:
+            project_id = self.remote_profile.get("project_id")
+
+            uid = uuid.uuid4()
+            path = os.path.join(str(project_id), str(uid), zip_filename)
+            setattr(job, "upload_path", path)
+            return path
+
+    def upload_job(self, job, common_files=None):
+        if common_files is None:
+            common_files = []
+        self.machine.gen_local_script(job)
+        zip_filename = job.job_hash + ".zip"
+        zip_task_file = os.path.join(self.local_root, zip_filename)
+
+        upload_file_list = [
+            job.script_file_name,
+        ]
+
+        upload_file_list.extend(common_files)
+
+        for task in job.job_task_list:
+            for file in task.forward_files:
+                upload_file_list.append(os.path.join(task.task_work_path, file))
+
+        upload_zip = Util.zip_file_list(
+            self.local_root, zip_task_file, file_list=upload_file_list
+        )
+        project_id = self.remote_profile.get("project_id", 0)
+
+        data = self.job.create(
+            project_id=project_id,
+            name=self.remote_profile.get("job_name", "DP-GEN"),
+            group_id=self.jgid,
+        )
+        self.jgid = data["jobGroupId"]
+        token = data["token"]
+
+        object_key = os.path.join(data["storePath"], zip_filename)
```

Comment:
These typing errors should be addressed. If they are false positive, add `  # type: ignore` to the end of the line.